index.js

```diff
@@ -0,0 +1,306 @@
+            if (caller.arguments && caller.arguments[0] && caller.arguments[0].target) {
+                currentEvent = caller.arguments[0];
+                log('The function at the bottom of the stack has an expected type. The current event is:', currentEvent);
+            }
+        } catch (e) {
+            log('Getting event from Function.caller failed, due to an error:', e);
+        }
+    } else {
+        log('window.event exists, of which the value is:', currentEvent);
+    }
+};
+
+/**
+ * @param {*} event Optional argument, an event to test with. Default value is currentEvent.
+ * @return {boolean} True if the event is legit, false if it is something that we should not allow window.open or dispatchEvent.
+ */
+var verifyEvent = function(event) {
+    event = event || (retrieveEvent(), currentEvent);
+    if (event) {
+        log('Verifying event');
+        var currentTarget = event.currentTarget;
+        if (currentTarget) {
+            var tagName = currentTarget.nodeName.toLowerCase();
+            if (tagName == '#document' || tagName == 'html' || tagName == 'body') {
+                log('VerifyEvent - the current event handler is suspicious, for the current target is either document, html, or body.');
+                return false;
+            } else if (maybeOverlay(currentTarget)) {
+                log('VerifyEvent - the current event handler is suspicious, for the current target looks like an artificial overlay.');
+                return false;
+            }
+        }
+    }
+    return true;
+};
+
+/************************************************************************************/
+/**
+ * Detects common overlay pattern.
+ * @param {Element} el an element to check whether it is an overlay.
+ * @return {boolean} true if el is an overlay.
+ */
+function maybeOverlay(el) {
+    var w = window.innerWidth, h = window.innerHeight;
+    if (el.offsetLeft << 4 < w && (w - el.offsetWidth) << 3 < w
+        && el.offsetTop << 4 < h && (h - el.offsetHeight) << 3 < w) {
+        var style = getComputedStyle(el);
+        var position = style.getPropertyValue('position');
+        var zIndex = style.getPropertyValue('z-index');
+        log('An element passed offset test.');
+        if ((position == 'fixed' || position == 'absolute') && zIndex > 1000) {
+            log('An element passed computedStyle test.');
+            return true;
+        }
+    }
+    // ToDo: the element may have been modified in the event handler.
+    // We may still test it using the inline style attribute.
+    return false;
+}
+
+/**
+ * Some popup scripts adds transparent overlays on each of page's links
+ * which disappears only when popups are opened.
+ * To restore the expected behavior, we need to detect if the event is 'masked' by artificial layers
+ * and redirect it to the correct element.
+ * ToDo: touch events: https://developer.mozilla.org/en/docs/Web/API/Touch_events
+ */
+var dispatchIfBlockedByMask = function() {
```

Comment:
Again, you should pass event as a function argument instead of using `currentEvent`.

Then it'd be easy to test that function:
```
var event = new MouseEvent(...);
// Stubs - use them to check how function worked
var _stopPropagation = { call: function() {  } };
var _dispatchEvent = { call: function(){} }
dispatchIfBlockedByMask(event);
// now assert that functions were (or were not) called
```